src/main/java/org/folio/innreach/domain/service/impl/MaterialTypeMappingServiceImpl.java

```diff
@@ -0,0 +1,115 @@
+import org.folio.innreach.domain.service.MaterialTypeMappingService;
+import org.folio.innreach.dto.MaterialTypeMappingDTO;
+import org.folio.innreach.dto.MaterialTypeMappingsDTO;
+import org.folio.innreach.mapper.MaterialTypeMappingMapper;
+import org.folio.innreach.repository.MaterialTypeMappingRepository;
+
+@RequiredArgsConstructor
+@Service
+@Transactional
+public class MaterialTypeMappingServiceImpl implements MaterialTypeMappingService {
+
+  private final MaterialTypeMappingRepository repository;
+  private final MaterialTypeMappingMapper mapper;
+
+  
+  @Override
+  @Transactional(readOnly = true)
+  public MaterialTypeMappingsDTO getAllMappings(UUID centralServerId, Integer offset, Integer limit) {
+    var example = mappingExampleWithServerId(centralServerId);
+
+    var totalCount = repository.count(example);
+
+    var result = new MaterialTypeMappingsDTO();
+    if (totalCount > 0) {
+      Page<MaterialTypeMapping> mappings = repository.findAll(example, PageRequest.of(offset, limit));
+
+      result.setMappings(mappings.stream()
+          .map(mapper::mapToDTO)
+          .collect(toList()));
+    }
+    result.setTotalRecords((int) totalCount);
+
+    return result;
+  }
+
+  @Override
+  @Transactional(readOnly = true)
+  public MaterialTypeMappingDTO getMapping(UUID centralServerId, UUID id) {
+    var mapping = findMapping(centralServerId, id);
+
+    return mapper.mapToDTO(mapping);
+  }
+
+  @Override
+  public MaterialTypeMappingDTO createMapping(UUID centralServerId, MaterialTypeMappingDTO dto) {
+    var entity = mapper.mapToEntity(dto);
+    entity.setCentralServer(centralServerRef(centralServerId));
+
+    var saved = repository.save(entity);
+
+    return mapper.mapToDTO(saved);
+  }
+
+  @Override
+  public MaterialTypeMappingDTO updateMapping(UUID centralServerId, UUID id,
+      MaterialTypeMappingDTO dto) {
+    var mapping = findMapping(centralServerId, id);
+
+    mapping.setMaterialTypeId(dto.getMaterialTypeId());
+    mapping.setCentralItemType(dto.getCentralItemType());
+
+    return mapper.mapToDTO(mapping);
+  }
+
+  @Override
+  public void deleteMapping(UUID centralServerId, UUID id) {
+    MaterialTypeMapping mapping = findMapping(centralServerId, id);
+    repository.delete(mapping);
+  }
+
+  private MaterialTypeMapping findMapping(UUID centralServerId, UUID id) {
+    return repository.findOne(mappingExampleWithServerIdAndId(centralServerId, id))
+        .orElseThrow(() -> new EntityNotFoundException("Material type mapping not found: id = " + id +
+            ", centralServerId = " + centralServerId));
+  }
+
+  private static Example<MaterialTypeMapping> mappingExampleWithServerId(UUID centralServerId) {
+    var toFind = new MaterialTypeMapping();
+    toFind.setCentralServer(centralServerRef(centralServerId));
+
+    return Example.of(toFind);
+  }
+
+  private static Example<MaterialTypeMapping> mappingExampleWithServerIdAndId(UUID centralServerId, UUID id) {
```

Comment:
Does this method have to be static?

Reply:
I tend to create utility methods as static. It has no relation with the object state, so can be "unlinked" from it. Also it's has better readability for me in the code that uses it
So it's just my preference